### ISSUE TITLE
Video player bug?

### ISSUE THREAD

**Issue by xumo** [cinder/Cinder#1114]
Whe compiling in Yosemite, XCode 6.4 and the master branch (glNext) and then playing it in a mac mini OS 10.8.3 with intel HD 4000 graphics cards the video draws a white square.
The same code with cinder 8.6 works fine.

Maybe is a AVFoundation framework issue but the mac AVFoundation marks 10.8 as supported, and the de console shows an AVFoundation message:
AVF info: hasOfflineRenderer

**Comment by andrewfb** [comment]
Hi - can you test your app on a different machine so we can try to isolate whether it's hardware specific?

**Comment by xumo** [comment]
I tested it on 4 mac mini late 2012
4GB Ram
Intel Graphics 4000 512mb
OS 10.8.3

I tested with my main app, and it runned well in only one. Then a tried an simple video player app and in none of them worked correctly.

I've done some app with video content in more recent mac mini (Yosemite) and it works ok. I could just upgrade it to Yosemite but the mac minis are in another city, i just brought on with me for testing.

**Comment by andrewfb** [comment]
OK - Is the symptom dependent on the codec? What does the samples/QuickTimeBasic do with the same file?

**Comment by xumo** [comment]
It is the same with samples/QuickTimeBasic. It draws a white rectangle.

**Comment by andrewfb** [comment]
What codec is it? Can QuickTimeBasic play any files at all on that machine? There are a number of codecs that the now-deprecated QuickTime could play which AVFoundation cannot.

**Comment by xumo** [comment]
H264, the audio AAC plays well.

Now I'll try with animation Codec

**Comment by xumo** [comment]
It doesn't play animation codec either

**Comment by ryanbartley** [comment]
Animation Codec isn't supported by AV Foundation.

**Comment by xumo** [comment]
I'll try more codecs then

**Comment by ryanbartley** [comment]
Find the list of deprecated codecs [here](https://forums.creativecow.net/thread/71/863688).

**Comment by xumo** [comment]
Tried with Apple ProRes, DV, h264 and it still blank. I'll do more test, but I think is not a codec issue.

**Comment by xumo** [comment]
The AVSimplePlayer example work fine with the h264 video. https://developer.apple.com/library/mac/samplecode/AVSimplePlayerOSX/Introduction/Intro.html#//apple_ref/doc/uid/DTS40011060 I'll try to rewrite it in 8.6

**Comment by andrewfb** [comment]
OK - would you be able to post the video somewhere, or at least one which exhibits the same issue?

**Comment by xumo** [comment]
This is the video
http://rodrigotorres.net/archivos/Video2.mp4

**Comment by pizthewiz** [comment]
I tried building QuickTimeBasic on OS X 10.8.5 / Xcode 5.1.1 and hit a link error in Boost::filesystem with a missing `_fchmodat` symbol - this seems to be a [known issue](https://svn.boost.org/trac/boost/ticket/10591) and is fixed in 1.60 but we'll otherwise need to apply [the patch](https://svn.boost.org/trac/boost/attachment/ticket/10591/boost-fs-fchmodat-darwin.patch) on top of 1.58. 😖

**Comment by pizthewiz** [comment]
Spun the `fchmodat` linking issue off as #1116 - patched and built `boost::filesystem` myself and also ran into the all white texture with Cinder/QuickTimeBasic built and run in an OS X 10.8.5 / Xcode 5.1.1 VM - I used an H.264 sample file provided via Apple's aging [QuickTime Sample files](https://support.apple.com/en-us/HT201549) page, but it might be helpful to have our own sample in _samples/data_ no?

**Comment by andrewfb** [comment]
Quick update, I was able to run this on a recent MacBook Pro running 10.10. I'll try to track down more similar hardware to the machine you're seeing it manifest on.

**Comment by xumo** [comment]
I tried building the filesystem from boost an replacing the libboost_filesystem.a but no luck.
I think updating to Yosemite would be more easy, or do you think is worth to keep digging into this Andrew?

**Comment by pizthewiz** [comment]
Right, a patched `boost::filesystem` is required to build on OS X 10.8 and 10.9, but it doesn't resolve (unrelated, just the same OS version) this video issue.
